catkin_home/src/arm_vision/include/ProcessingCloud.h

```diff
@@ -0,0 +1,419 @@
+                extract.setNegative(false);
+
+                // Get the points associated with the planar surface
+                extract.filter(*cloud_plane);
+                cloudsplane.push_back(cloud_plane);
+
+                //Save the coeffiecients of the plane equation
+                for (int k = 0; k < 4; k++) {
+                        vectorcoefficients[i].push_back(coefficients->values[k]);
+                }
+                //std::cout << "PointCloud representing the planar component: " <<
+cloud_plane->points.size() << " data points." << std::endl;
+                npointsplane.push_back(cloud_plane->points.size());
+
+
+                // Remove the planar inliers, extract the rest
+                extract.setNegative(true);
+                extract.filter(*cloud_f);
+                *RawCloud = *cloud_f;
+                i++;
+        }
+        i = 0;
+        //find the maximum planar cloud
+        maxindexp;
+        int countp = 0;
+        if (npointsplane.size() == 0) {
+                std::cout << "We have not found any cloud";
+        }
+        else {
+                unsigned int maxpointsp = npointsplane[0];
+                maxindexp = 0;
+                for (std::vector<unsigned int>::iterator it = npointsplane.begin(); it !=
+npointsplane.end(); ++it) { if (*it > maxpointsp) { maxpointsp = *it; maxindexp = i;
+                        }
+                        i++;
+                }
+        }
+        return vectorcoefficients;
+}*/
+float dotp(pcl::PointXYZRGB v1, pcl::PointXYZRGB v2) {
+    float product = 0;
+    product += v1.x * v2.x;
+    product += v1.y * v2.y;
+    product += v1.z * v2.z;
+    return product;
+}
+pcl::PointCloud<pcl::PointXYZRGB>::Ptr PlanarRANSAC(
+    const pcl::PointCloud<pcl::PointXYZRGB>::Ptr cloud, double &zPLANE) {
+    // perform ransac planar filtration to remove table top
+    pcl::ModelCoefficients::Ptr coefficients(new pcl::ModelCoefficients);
+    pcl::PointIndices::Ptr inliers(new pcl::PointIndices);
+    // Create the segmentation object
+    pcl::SACSegmentation<pcl::PointXYZRGB> seg1;
+    // Optional
+    seg1.setOptimizeCoefficients(true);
+    // Mandatory
+    seg1.setModelType(pcl::SACMODEL_PLANE);
+    seg1.setMethodType(pcl::SAC_RANSAC);
+    seg1.setDistanceThreshold(0.01);
+
+    seg1.setInputCloud(cloud);
+    seg1.segment(*inliers, *coefficients);
+
+    // Create the filtering object
+    pcl::ExtractIndices<pcl::PointXYZRGB> extract;
+    pcl::PointCloud<pcl::PointXYZRGB>::Ptr outcloud_XYZ(new pcl::PointCloud<pcl::PointXYZRGB>);
+
+    extract.setInputCloud(cloud);
+    extract.setIndices(inliers);
+    extract.setNegative(true);
+    extract.filter(*outcloud_XYZ);
+
+    pcl::PointIndices::Ptr inliers2(new pcl::PointIndices);
+    pcl::PointXYZRGB dumb;
+    dumb.x = cloud->points[inliers->indices[0]].x;
+    dumb.y = cloud->points[inliers->indices[0]].y;
+    dumb.z = cloud->points[inliers->indices[0]].z;
```

Comment:
Is it possible to pass x,y,z as parameters to dumb in its instantiation? instead of assigning them afterwards

Reply:
probably should name "dumb" something more descriptive